src/containers/containers.c

```diff
@@ -373,123 +374,13 @@ roaring_container_iterator_t container_init_iterator_last(const container_t *c,
 bool container_iterator_next(const container_t *c, uint8_t typecode,
                              roaring_container_iterator_t *it,
                              uint16_t *value) {
-    switch (typecode) {
-        case BITSET_CONTAINER_TYPE: {
-            const bitset_container_t *bc = const_CAST_bitset(c);
-            it->index++;
-
-            uint32_t wordindex = it->index / 64;
-            if (wordindex >= BITSET_CONTAINER_SIZE_IN_WORDS) {
-                return false;
-            }
-
-            uint64_t word =
-                bc->words[wordindex] & (UINT64_MAX << (it->index % 64));
-            // next part could be optimized/simplified
-            while (word == 0 &&
-                   (wordindex + 1 < BITSET_CONTAINER_SIZE_IN_WORDS)) {
-                wordindex++;
-                word = bc->words[wordindex];
-            }
-            if (word != 0) {
-                it->index = wordindex * 64 + roaring_trailing_zeroes(word);
-                *value = it->index;
-                return true;
-            }
-            return false;
-        }
-        case ARRAY_CONTAINER_TYPE: {
-            const array_container_t *ac = const_CAST_array(c);
-            it->index++;
-            if (it->index < ac->cardinality) {
-                *value = ac->array[it->index];
-                return true;
-            }
-            return false;
-        }
-        case RUN_CONTAINER_TYPE: {
-            if (*value == UINT16_MAX) {  // Avoid overflow to zero
-                return false;
-            }
-
-            const run_container_t *rc = const_CAST_run(c);
-            uint32_t limit =
-                rc->runs[it->index].value + rc->runs[it->index].length;
-            if (*value < limit) {
-                (*value)++;
-                return true;
-            }
-
-            it->index++;
-            if (it->index < rc->n_runs) {
-                *value = rc->runs[it->index].value;
-                return true;
-            }
-            return false;
-        }
-        default:
-            assert(false);
-            roaring_unreachable;
-            return false;
-    }
+    return container_iterator_next_inline(c, typecode, it, value);
```

Comment:
What's the benefit of using a separate `_inline` function rather than defining `container_iterator_next` as inline in `containers.h`?

Reply:
Excellent idea. Let us do it.